### PR TITLE
feat: bot-aware query scoping and dashboard toggles (#26, #46, #47)

### LINKED ISSUE #26: Add bot-aware query scoping to AnalyticsQuery
Parent: #20

## Task

Update the `AnalyticsQuery` service and related query methods to exclude bot traffic by default, with an option to include it.

### Changes

- All dashboard queries should join/filter on `visitors.is_bot = false` by default
- Add `includeBots()` / `onlyBots()` / `withBots()` query modifiers
- Update `AnalyticsQuery` service methods to apply bot filtering
- Update Livewire dashboard components to use bot-filtered queries

### Files to modify

- `src/Services/AnalyticsQuery.php` — Add bot filtering to all query methods
- `src/Http/Livewire/Widgets/StatsCards.php` — Use filtered queries
- `src/Http/Livewire/Widgets/VisitorsChart.php` — Use filtered queries
- `src/Http/Livewire/Widgets/TopPages.php` — Use filtered queries
- `src/Http/Livewire/Widgets/TrafficSources.php` — Use filtered queries
- `src/Http/Livewire/Widgets/RealtimeVisitors.php` — Use filtered queries

### Acceptance Criteria

- [ ] All dashboard queries exclude bot visitors by default
- [ ] `includeBots()` shows all traffic (human + bot)
- [ ] `onlyBots()` shows only bot traffic
- [ ] Query performance is not significantly impacted (indexed column)
- [ ] API query endpoints also respect bot filtering
- [ ] Tests verify filtering behavior

### LINKED ISSUE #46: Add bot-filter awareness to React dashboard components
Parent: #20
Related: #26 (backend query scoping + Livewire widgets)

## Task

Wire the React analytics dashboard components to respect bot filtering, mirroring the Livewire changes in #26. The bot-aware `AnalyticsQuery` (#26) makes the `/api/analytics/*` endpoints exclude bots by default, so React components should consume the filtered data and expose the same include/view-bots control as the Livewire dashboard.

### Changes

- Ensure all React dashboard components consume the bot-filtered API responses (excluded by default)
- Add an "include bot traffic" toggle to the React dashboard that re-fetches with the API's `withBots`/`includeBots` parameter
- Apply to: StatsCards, VisitorsChart, TopPages, TrafficSources, RealtimeVisitors React equivalents

### Files

- React dashboard components under `resources/js/react/`

### Acceptance Criteria

- [ ] React dashboard excludes bot traffic by default (parity with Livewire)
- [ ] Include-bots toggle re-fetches and updates all widgets
- [ ] Behavior matches the Livewire dashboard (#26)
- [ ] Accessible and responsive, consistent with other React components

### LINKED ISSUE #47: Add bot-filter awareness to Vue dashboard components
Parent: #20
Related: #26 (backend query scoping + Livewire widgets)

## Task

Wire the Vue 3 analytics dashboard components to respect bot filtering, mirroring the Livewire changes in #26 and the React equivalent (#46). Build after the React version for parity. The bot-aware `AnalyticsQuery` (#26) makes the `/api/analytics/*` endpoints exclude bots by default, so Vue components should consume the filtered data and expose the same include/view-bots control as the Livewire dashboard.

### Changes

- Ensure all Vue dashboard components consume the bot-filtered API responses (excluded by default)
- Add an "include bot traffic" toggle to the Vue dashboard that re-fetches with the API's `withBots`/`includeBots` parameter
- Apply to: StatsCards, VisitorsChart, TopPages, TrafficSources, RealtimeVisitors Vue equivalents

### Files

- Vue dashboard components under `resources/js/vue/`

### Acceptance Criteria

- [ ] Vue dashboard excludes bot traffic by default (parity with Livewire)
- [ ] Include-bots toggle re-fetches and updates all widgets
- [ ] Behavior matches the Livewire dashboard (#26) and React version
- [ ] Accessible and responsive, consistent with other Vue components

### PR DESCRIPTION
**Pull request by ViewFromTheBox**
## Description

Excludes confirmed bot traffic from analytics data by default across the Livewire, React, and Vue dashboards, with an opt-in toggle to include or isolate bot traffic. Implements the three parallel sub-issues of #20 together since they share a single API contract (`bots=exclude|include|only`).

**Closes:** #26

## Type of Change

- [x] New feature (adds new functionality)

## Related Issue

**Issue:** #26 (also closes #46, #47; parent #20)

## Motivation and Context

AI crawler and bot traffic inflates analytics. #26 adds backend query scoping so dashboards show real human traffic by default; #46 and #47 surface the same control in the React and Vue dashboards. They share the `bots` API parameter, so they were built together for a consistent contract.

## Changes Made

- `LocalAnalyticsProvider::applyBotFilter()` wired into `applyFilters` (inherited by all query methods) and realtime visitors. Exclude mode removes only **confirmed** bots, so rows with an unresolved visitor stay counted.
- `AnalyticsQuery` gains `includeBots()` / `withBots()` / `onlyBots()` / `excludeBots()` modifiers; bot mode is folded into filters so cache keys stay distinct. Default is exclude.
- `AnalyticsQueryController` reads `?bots=exclude|include|only` so API endpoints respect the filter.
- Livewire: `WithAnalyticsWidget` adds an `includeBots` flag + `analytics-bots-toggled` listener; `AnalyticsDashboard::toggleBots()` plus a header toggle button.
- React/Vue: shared `BotFilterMode` type + `bots` query param; "Include bot traffic" toggle on both dashboard pages emitting a host callback/event (mirrors the existing date-range pattern).

## How Has This Been Tested?

**Testing Environment:**
- Operating System: macOS
- Browser (if applicable): manual dashboard verification
- PHP Version: 8.4
- Project Version: 1.2 (release/1.2)

**Tests Performed:**
1. Full Pest suite (516 passed, 2 skipped).
2. New `BotAwareQueryScopingTest` covering default exclude, include, only, per-call reset, orphan-visitor regression, sessions, device breakdown, realtime.
3. Manual browser check of the dashboard toggle flipping counts.

## Accessibility Tests Run

- [x] Keyboard navigation tested
- [x] ARIA labels checked

**Details:** Toggle controls use native checkbox inputs with `aria-label`/`aria-pressed`; consistent with existing dashboard controls.

## Tests Added

- [x] Unit/feature tests added/updated
- [x] All tests passing

**Test details:** `tests/Feature/BotAwareQueryScopingTest.php` (13 tests). No JS test harness exists in this package, so JS changes are covered by the build/types.

## Documentation

- [x] Inline code documentation added

**Documentation details:** PHPDoc/TSDoc on all new methods, props, and types.

## Pre-Submission Checklist

- [x] Code passes all tests
- [x] Code has been linted (php-cs-fixer)
- [x] Code follows project style guide
- [x] Self-review completed
- [x] No new warnings generated

## Summary by CodeRabbit

* **New Features**
  * Added an "Include bot traffic" toggle to the analytics dashboard with modes to exclude, include, or show only bot traffic; dashboard metrics and realtime counts respect this setting.
* **Tests**
  * Added feature tests validating bot-aware analytics behavior, realtime counts, and edge cases.
* **Chores**
  * Backend analytics and realtime endpoints updated to honor bot-filtering options.

[Review Change Stack](https://app.coderabbit.ai/change-stack/ArtisanPack-UI/analytics/pull/57?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: src/Providers/LocalAnalyticsProvider.php [hunk 1]
```diff
@@ -633,6 +634,49 @@ protected function applyFilters( Builder $query, array $filters, ?string $tableP
...
+            return $query;
+        }
+
+        $model = $query->getModel();
+
+        if ( $model instanceof Visitor ) {
+            $query->where( $model->getTable() . '.is_bot', 'only' === $mode );
+
+            return $query;
+        }
```
[the remark this answers is not in the document]
Comment (ViewFromTheBox) [reply src/Providers/LocalAnalyticsProvider.php:676]: Skipping this one: `analytics_visitors.is_bot` is defined as `boolean('is_bot')->default(false)` (NOT NULL) in the migration, and the column is added with that default applied to existing rows. So NULL `is_bot` can't occur, and `where('is_bot', false)` correctly keeps all non-bots in exclude mode. Per project guidelines I'm not adding handling for a state the schema makes impossible.
